File: src/main/java/um/haberes/rest/model/Codigo.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Codigo extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 2630433768400508483L;

	@Id
	private Integer codigoId;

	private String nombre = "";
	private Byte docente = 0;
	private Byte noDocente = 0;
	private Byte transferible = 0;

}

File: src/main/java/um/haberes/rest/kotlin/model/Codigo.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable

@Entity
@Table
data class Codigo(

    @Id
    var codigoId: Int? = null,
    var nombre: String = "",
    var docente: Byte = 0,
    var noDocente: Byte = 0,
    var transferible: Byte = 0,
    var incluidoEtec: Byte = 0,
    var afipConceptoSueldoIdPrimerSemestre: Long? = null,
    var afipConceptoSueldoIdSegundoSemestre: Long? = null,

    @OneToOne
    @JoinColumn(name = "afipConceptoSueldoIdPrimerSemestre", insertable = false, updatable = false)
    var afipConceptoSueldoPrimerSemestre: AfipConceptoSueldo? = null,

    @OneToOne
    @JoinColumn(name = "afipConceptoSueldoIdSegundoSemestre", insertable = false, updatable = false)
    var afipConceptoSueldoSegundoSemestre: AfipConceptoSueldo? = null

) : Auditable()
